src/array/ops/broadcast.rs

```diff
@@ -205,6 +205,56 @@ impl Broadcastable for FixedSizeListArray {
     }
 }
 
+impl Broadcastable for StructArray {
+    fn broadcast(&self, num: usize) -> DaftResult<Self> {
+        if self.len() != 1 {
+            return Err(DaftError::ValueError(format!(
+                "Attempting to broadcast non-unit length Array named: {}",
+                self.name()
+            )));
+        }
+        let arrow_arr: &arrow2::array::StructArray = self.downcast();
+        let arrays = arrow_arr
+            .values()
+            .iter()
+            .map(|field_arr| {
+                arrow2::compute::concatenate::concatenate(
+                    std::iter::repeat(field_arr.as_ref())
+                        .take(num)
+                        .collect::<Vec<&dyn arrow2::array::Array>>()
+                        .as_slice(),
+                )
+            })
+            .collect::<Result<Vec<Box<dyn arrow2::array::Array>>, _>>()?;
+        let validity = arrow_arr.validity().map(|v| {
+            arrow2::bitmap::Bitmap::from_iter(
+                std::iter::repeat(v.iter().next().unwrap())
+                    .take(num)
+                    .collect::<Vec<bool>>(),
```

Comment:
I dont think you need to materialize as a Vec if you are doing `Bitmap::from_iter `